### PR TITLE
Fixes #25975 - Add ability to read compute_attributes via foreman API

### PR DESCRIPTION
**Pull request by bagasse**
Add ability to read compute_attribute to foreman API.

This is needed for [compute attribute foreman-ansible-modules support](https://github.com/theforeman/foreman-ansible-modules/issues/222)

### CODE REVIEW

File: app/controllers/api/v2/compute_attributes_controller.rb [hunk 1]
```diff
@@ -3,14 +3,22 @@ module V2
...
-      before_action :find_resource, :only => :update
+      before_action :find_resource, :only => [:show, :update]
 
       def_param_group :compute_attribute do
         param :compute_attribute, Hash, :required => true, :action_aware => true do
           param :vm_attrs, Hash, :required => true
         end
       end
 
+      api :GET, "/compute_resources/:compute_resource_id/compute_profiles/:compute_profile_id/compute_attributes", N_("Show a compute attributes set")
```
Comment (xprazak2) [inline app/controllers/api/v2/compute_attributes_controller.rb]: should be "/compute_resources/:compute_resource_id/compute_profiles/:compute_profile_id/compute_attributes/:id", same for the others below

**Comment by xprazak2** [comment]
Looking good, apipie doc strings just need a small change. Could you add tests please? Would it make sense to add `index` action into the controller as well?

**Comment by bagasse** [comment]
[test prprocessor]

**Comment by bagasse** [comment]
@xprazak2, done I added index too.

File: app/controllers/api/v2/compute_attributes_controller.rb [hunk 2]
```diff
@@ -1,16 +1,42 @@
 module Api
   module V2
     class ComputeAttributesController < V2::BaseController
+      include Api::Version2
       include Foreman::Controller::Parameters::ComputeAttribute
+      include ParameterAttributes
```
Comment (xprazak2) [inline app/controllers/api/v2/compute_attributes_controller.rb]: I do not see why we `include ParameterAttributes`. What am I missing?

**Comment by xprazak2** [comment]
Looks good and works nicely. @bagasse, why do we need to import `ParameterAttributes` into `ComputeAttributesController`?

File: app/controllers/api/v2/compute_attributes_controller.rb (hunk 2 above)
Comment (bagasse) [reply app/controllers/api/v2/compute_attributes_controller.rb]: That's not needed. removed. When I was adding index, I took a look on subnet API controller to find how nested resources and scoped search were working and I added this but it' not needed. Thanks for the pointer.

**Comment by bagasse** [comment]
[test foreman]

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks!

**Pull request merged**